Classes/ImagineFactory.php

```diff
@@ -37,41 +38,86 @@ class ImagineFactory extends AbstractImagineFactory
      * \Imagine\Gd\Image.
      *
      * @param string $className If specified, this factory will create an instance of the driver dependent class
+     * @param mixed ...$arguments
+     * @return \Imagine\Image\ImagineInterface This is not necessarily true depending if you give another className
+     * @api
+     * @deprecated Use createDriver or createImagineClass as needed
+     * @see createDriver
+     * @see createImagineClass
+     */
+    public function create($className = 'Imagine', ...$arguments)
+    {
+        $this->configureDriverSpecificSettings($this->settings['driver']);
+        return $this->createImagineClass($this->settings['driver'], $className, ...$arguments);
+    }
+
+    /**
+     * Factory method to create a ImagineInterface driver instance, if you just want the currently configured one,
+     * inject the \Imagine\Image\ImagineInterface which will use this factory method with the configured driver class.
+     *
+     * This will also apply driver specific settings.
+     *
+     * @param string $driverName
+     * @param string $className
+     * @param mixed ...$arguments
      * @return \Imagine\Image\ImagineInterface
-     * @throws \ReflectionException
      * @api
      */
-    public function create($className = 'Imagine')
+    public function createDriver(string $driverName, ...$arguments): ImagineInterface
     {
-        $this->configureDriverSpecificSettings();
+        $this->configureDriverSpecificSettings($driverName);
+        $className = $this->buildImagineClassName($driverName);
 
-        $className = 'Imagine\\' . $this->settings['driver'] . '\\' . $className;
-        $arguments = array_slice(func_get_args(), 1);
+        return new $className(...$arguments);
+    }
 
-        switch (count($arguments)) {
-            case 0: $object = new $className(); break;
-            case 1: $object = new $className($arguments[0]); break;
-            case 2: $object = new $className($arguments[0], $arguments[1]); break;
-            case 3: $object = new $className($arguments[0], $arguments[1], $arguments[2]); break;
-            case 4: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3]); break;
-            case 5: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4]); break;
-            case 6: $object = new $className($arguments[0], $arguments[1], $arguments[2], $arguments[3], $arguments[4], $arguments[5]); break;
-            default:
-                $class = new \ReflectionClass($className);
-                $object =  $class->newInstanceArgs($arguments);
-        }
```

Comment:
:eyes:

good to have that gone

Reply:
welcome to how our object manager looked a few years back 😂 It was faster though because the reflection is slow.